restclients_core/dao.py

```diff
@@ -178,6 +198,31 @@ def _load_resource(self, method, url, headers, body):
 
         return response
 
+    def prometheus_duration(self, duration):
+        """
+        Override this method if you have service-specific logic
+        around response times
+        """
+        self.prometheus_duration_observation(duration)
+
+    def prometheus_status(self, response):
+        """
+        Override this method to insert service-specific logic
+        before setting the response status code observation
+
+        e.g., If the service applies special meaning to, say, 404 response
+        that might not make sense to observe
+        """
+        self.prometheus_status_observation(response.status)
+
+    def prometheus_duration_observation(self, duration):
+        prometheus_duration.labels(self.service_name()).observe(duration)
+
+    def prometheus_status_observation(self, status):
+        # status category buckets
+        prometheus_status.labels(self.service_name()).observe(
+            (status / 100) * 100)
```

Comment:
Will "/" work here, or does it need to be "//" for it to be floor division? I remember reading something about how there was a switch from python 2 to 3, I'll look for it.

Reply:
"Here is another surprise, In Python 3, ‘/’ operator does floating point division for both int and float arguments."  - from: https://www.geeksforgeeks.org/division-operator-in-python/

ezturner@ketchup:~$ python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> 1 / 2
0.5
>>> 1 // 2
0

Reply:
// for the win... dev pr waiting